Source/CarthageKit/Project.swift

```diff
@@ -230,15 +230,32 @@ public final class Project { // swiftlint:disable:this type_body_length
 			.startOnQueue(cloneOrFetchQueue)
 	}
 
-	func downloadBinaryFrameworkDefinition(binary: BinaryURL) -> SignalProducer<BinaryProject, CarthageError> {
+    func downloadBinaryFrameworkDefinition(binary: BinaryURL) -> SignalProducer<BinaryProject, CarthageError> {
 		return SignalProducer<Project.CachedBinaryProjects, CarthageError>(value: self.cachedBinaryProjects)
 			.flatMap(.merge) { binaryProjectsByURL -> SignalProducer<BinaryProject, CarthageError> in
 				if let binaryProject = binaryProjectsByURL[binary.url] {
 					return SignalProducer(value: binaryProject)
 				} else {
-					self._projectEventsObserver.send(value: .downloadingBinaryFrameworkDefinition(.binary(binary), binary.url))
-
-					return URLSession.shared.reactive.data(with: URLRequest(url: binary.url))
+                    self._projectEventsObserver.send(value: .downloadingBinaryFrameworkDefinition(.binary(binary), binary.url))
+                    
+                    var request = URLRequest(url: binary.url)
+                    
+                    // When downloading a binary, `carthage` will take into account the user's
+                    // `~/.netrc` file to determine authentication credentials
+                    switch Netrc.load() {
+                    case .success(let machines):
+                        if let index = machines.firstIndex(where: { $0.name == binary.url.host }) {
+                            let machine = machines[index]
+                            let authString = "\(machine.login):\(machine.password)"
+                            if let authData = authString.data(using: .utf8) {
+                                let authValue = "Basic \(authData.base64EncodedString())"
+                                request.addValue(authValue, forHTTPHeaderField: "Authorization");
+                            }
```

Comment:
I'd take `Netrc` an actual type and then move this logic for finding the credentials into a method on it.

Reply:
Thanks for the suggestion 💪 Just to double check that we are on the same page - basically are you thinking about adding a method to `Netrc` struct like:

```Swift
// Netrc.swift

public func credentials(for machine: String) -> NetrcCredentials? {
    ...
}
```

Example usage:

``Swift
switch Netrc.load() {
                    case .success(let netrc):
                        let credentials = netrc.credentials(for: binary.url.host)
                        request.addValue(credentials.authorization, forHTTPHeaderField: "Authorization");
```

WDYT? 🤔

Reply:
Yup 👍 

But:

1. I'd probably pass the entire `URL` as the input to the method
2. I might return the authorization value directly

```
case let .success(netrc):
  if let authorization = netrc.authorizatino(for: binary.url) {
    request.addValue(authorization, forHTTPHeaderField: "Authorization")
```

Reply:
The changes have been implemented @mdiep 🙇